olive/model/__init__.py

```diff
@@ -80,6 +82,15 @@ def model_path(self) -> str:
         """Return local model path."""
         return self.get_local_resource("model_path")
 
+    def extend_model_attributes(self, model_attr) -> Dict[str, Any]:
+        # self.model_attributes > model_attr, should respect self.model_attributes when there is conflict
```

Comment:
It may not be an extension all the time.

Reply:
update should be better?

Reply:
IMHO, if the priority of hf_config/model_attr is different, the pass itself should be responsible do what ever merge and finally set the model_attr. 

From this point of view, a directly assignment is enough.

Reply:
Basically, the priority logic should not be responsibility of model. Instead, it should be in another place.

Reply:
`extend_model_attributes` with the priority order: self.model_attributes > model_attr
for me, the `update_model_attributes` with the priority order: self.model_attributes > model_attr
maybe we can add method `update_model_attributes` in case any pass requires in the future.

Reply:
personally, I don't like the model maintain the logic here.

Reply:
Updated to just run assignment after _run_for_config. We assume that once the pass modified the attributes, that is the final attributes. We should not overwrite it anymore with input model's attr.